src/test/java/io/github/fbiville/nestedrelationquerysdn6/NestedRelationQuerySdn6ApplicationTests.java

```diff
@@ -14,65 +16,133 @@
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
 
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 @Testcontainers
 @DataNeo4jTest
+@TestInstance(Lifecycle.PER_CLASS)
 class NestedRelationQuerySdn6ApplicationTests {
 
     @Autowired
     private Driver driver;
 
-    @Container
-    private static final Neo4jContainer<?> neo4jContainer = new Neo4jContainer<>("neo4j:4.0");
+    //@Container
+    //private static final Neo4jContainer<?> neo4jContainer = new Neo4jContainer<>("neo4j:4.0");
 
     @DynamicPropertySource
     static void neo4jProperties(DynamicPropertyRegistry registry) {
-        registry.add("spring.neo4j.uri", neo4jContainer::getBoltUrl);
-        registry.add("spring.neo4j.authentication.username", () -> "neo4j");
-        registry.add("spring.neo4j.authentication.password", neo4jContainer::getAdminPassword);
+        //registry.add("spring.neo4j.uri", neo4jContainer::getBoltUrl);
+        registry.add("spring.neo4j.uri", () -> "bolt://localhost:7687");
+        //registry.add("spring.neo4j.authentication.username", () -> "neo4j");
+        //registry.add("spring.neo4j.authentication.password", neo4jContainer::getAdminPassword);
     }
 
-    @BeforeEach
-    void prepare() {
+    @BeforeAll
+    void prepare(@Autowired ChangesetRepository repository) {
         try (Session session = driver.session()) {
             session.writeTransaction(tx -> {
                 tx.run("MATCH (n) DETACH DELETE n");
-                tx.run("CREATE (:Changeset:Hypercube {USER: 'some-user'})-[:CHANGESET_IN]->(ws:Workspace {STATE: 'some-state'}), " +
-                        "     (ws)-[:WORKSPACE_IN]->(:Hypercube:Work {STATE: 'some-other-state'})");
+                //tx.run("CREATE (:Workspace:Hypercube {STATE: 'some-xxx', ID : 1 , CANCELED : false})<-[:CHANGESET_IN {VALID : true}]-(:Changeset:Hypercube {USER: 'some-user', ID: 1, CANCELED : false})-[:CHANGESET_IN {VALID : true}]->(ws:Workspace:Hypercube {STATE: 'some-state', ID : 1 , CANCELED : false}), " +
```

Comment:
if you could provide a small Cypher dump for the tests to run, that would be great

Reply:
I'll add some real data later, but so far this small dataset is enough to catch the issue tho :)

Reply:
the small dataset I initially defined does not match the updated entities anymore and I don't have a local DB with the data at hand ;(

Reply:
Just a question, aren't the new line 50 and 51 providing a small interaction that may work on a container?

Reply:
I completely missed these 😅